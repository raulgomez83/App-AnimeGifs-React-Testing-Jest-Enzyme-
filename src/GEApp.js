import React ,{useState} from 'react'
import { AddCategory } from './componentes/AddCategory'
import { GifGrid } from './componentes/GifGrid'

export const GEApp = ({defaultCategories=[]} ) => {

    const [categories, setCategories]=useState(defaultCategories)
       return (
        <>
            <h2 className="animate__animated animate__zoomIn">Anime GIF Searcher</h2>
            <AddCategory setCategories={setCategories} />

           <ul>
            { categories.map(category =>
            <GifGrid key={category} category={category}></GifGrid>)
            }
           </ul>
        </>
    )
}
