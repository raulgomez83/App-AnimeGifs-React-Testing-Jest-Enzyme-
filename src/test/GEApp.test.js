import React from "react"
import { GEApp } from "../GEApp";
import "@testing-library/jest-dom";

const { shallow } = require("enzyme");

describe('Testing GEApp', () => {

    test('should works', () => {
        const wrapper= shallow(<GEApp />);
        expect(wrapper).toMatchSnapshot();
    })
     test('should show categories list', () => {
        const categories=["Naruto","Boku no Hero"];
        const wrapper=shallow(<GEApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    })
})
