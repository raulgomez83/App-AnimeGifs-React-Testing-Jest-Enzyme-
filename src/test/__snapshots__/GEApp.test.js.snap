// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GEApp should show categories list 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__zoomIn"
  >
    Anime GIF Searcher
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Boku no Hero"
      key="Boku no Hero"
    />
  </ul>
</Fragment>
`;

exports[`Testing GEApp should works 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__zoomIn"
  >
    Anime GIF Searcher
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
